RocketChat/Rocket.Chat
# [IMPROVE] General room reappears after being deleted

## Proposed changes (including videos or screenshots)


 - Check the `Show_Setup_Wizard` Setting's value to control whether the general room should be created. This channel will only be created if the `Show_Setup_Wizard` Setting is 'pending'.



## Issue(s)

[Task - ClickUp](https://app.clickup.com/t/hx7dq7)

## Steps to test or reproduce

1. Log in as an administrator and delete the general room;
2. Go to Administration > General and turn off the "Keep general room on the server" setting;
3. Restart the server;
4. Go to Administration > Rooms and search for "general". **_Expected behavior:_** no results should be found (the room shouldn't exist after server restart).
**Note:** the tests should behave the same way using `TEST_MODE=true`.

## Further comments

The **`Show_Setup_Wizard`** setting's value is updated from 'pending' to 'in_progress' or 'completed' if there is at least one user in the server OR if the setup wizard is completed. Once it is updated from 'pending' to any other value, it'll never return to pending again (that's why it is safe to assume that the general channel will only be created once).
